src/main/java/org/candlepin/async/tasks/RevokeEntitlementsJob.java

```diff
@@ -0,0 +1,173 @@
+        this.activationKeyCurator = Objects.requireNonNull(activationKeyCurator);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void execute(JobExecutionContext context) throws JobExecutionException {
+        JobArguments jobArguments = context.getJobArguments();
+        String ownerKey = jobArguments.getAsString(OWNER_KEY);
+
+        log.info("{} has started for owner {}", JOB_KEY, ownerKey);
+        Owner owner = ownerCurator.getByKey(ownerKey);
+        if (owner == null) {
+            log.info("Unable to find owner for owner key \"{}\"", ownerKey);
+            context.setJobResult("Unable to find owner for owner key \"%s\"", ownerKey);
+            return;
+        }
+
+        int batchSize = getBatchSize();
+
+        int removedPools = activationKeyCurator.removeActivationKeyPools(ownerKey);
+        log.debug("Removed {} activation key pools", removedPools);
+
+        /*
+         * Revoke all entitlements for batches of consumers. Each batch is processed in a seperate
+         * transaction.
+         */
+        List<String> consumerUuids = consumerCurator.getSystemConsumerUuidsByOwner(owner.getKey());
+        int entRevokeCount = 0;
+        for (List<String> batch : Iterables.partition(consumerUuids, batchSize)) {
+            log.debug("Revoking entitlements for consumers with UUIDs: {}", batch);
+            entRevokeCount += poolService.revokeAllEntitlements(batch, false);
+        }
+
+        log.info("{} has run! {} entitlements revoked and {} activation key pools removed.", JOB_KEY,
+            entRevokeCount, removedPools);
+        String msg = "%s completed successfully. %d entitlements revoked and %d activation key pools " +
+            "removed.";
+        context.setJobResult(msg, JOB_NAME, entRevokeCount, removedPools);
+    }
+
+    /**
+     * Job configuration object for the {@link RevokeEntitlementsJob}
+     */
+    public static class RevokeEntitlementsJobConfig extends JobConfig<RevokeEntitlementsJobConfig> {
+
+        public RevokeEntitlementsJobConfig() throws JobExecutionException {
+            this.setJobKey(JOB_KEY)
+                .setJobName(JOB_NAME)
+                .addConstraint(JobConstraints.uniqueByArguments(OWNER_KEY))
+                .setRetryCount(MAX_RETRY_COUNT);
+        }
+
+        /**
+         * Sets the owner key for this {@link RevokeEntitlementsJobConfig}.
+         *
+         * @param ownerKey
+         *  the owner key to set
+         *
+         * @throws IllegalArgumentException
+         *  if the owner key is null or blank
+         *
+         * @return a reference to this RevokeEntitlementsJobConfig
+         */
+        public RevokeEntitlementsJobConfig setOwnerKey(String ownerKey) {
+            if (ownerKey == null || ownerKey.isBlank()) {
+                throw new IllegalArgumentException("owner key is null or blank");
+            }
+
+            this.setJobArgument(OWNER_KEY, ownerKey);
```

Comment:
in addition to adding the owner key as an argument here, we want to call `setContextOwner(owner)` on the JobConfig here (see other owner-scoped jobs for examples). that will allow us to to query for job statuses using the owner key

Reply:
Updated